browser/app/winlauncher/freestanding/LoaderPrivateAPI.h

```diff
@@ -54,8 +54,8 @@ class NS_NO_VTABLE LoaderPrivateAPI : public nt::LoaderAPI {
  */
 void EnsureInitialized();
 
-extern LoaderPrivateAPI& LoaderPrivateAPI();
-#define gLoaderPrivateAPI LoaderPrivateAPI()
+extern LoaderPrivateAPI& gLoaderPrivateAPI();
+// #define gLoaderPrivateAPI LoaderPrivateAPI()
```

Comment:
It seems like this macro would already have done the replacement you've manually applied with these changes, what was the error you were getting?

Reply:
I was getting a compiler (clang) error remarking that I needed to insert a `class` keyword before `LoaderPrivateAPI&` at both line 57 in `LoaderPrivateAPI.h` and line 123 in `LoaderPrivateAPI.cpp` like so:

```cpp
extern class LoaderPrivateAPI& LoaderPrivateAPI();
```
and
```cpp
class LoaderPrivateAPI& LoaderPrivateAPI() { ... }
```

The reason for the error is pretty obvious, since there's a collision for the name LoaderPrivateAPI. Though I wasn't terribly sure about _why_  adding class was being suggested until I did some research and learned about _type/non-type hiding_ ([source](https://en.cppreference.com/w/cpp/language/lookup)). Apparently, adding `class` is the C++ equivalent of adding `struct` in front of a struct name in typical C. But even after adding `class` to both lines, the `extern` in `LoaderPrivateAPI.h` failed to compile due to a syntax error. Seems type/non-type hiding can't be used everywhere. Hence replacing the macro with a name change. The name change resolves the type/non-type hiding problem by removing the non-type name from the equation.